tests/model_registry/rest_api/conftest.py

```diff
@@ -63,3 +92,250 @@ def updated_model_registry_resource(
+    request: pytest.FixtureRequest,
+) -> Generator[str, Any, Any]:
+    """
+    Patches the odh-trusted-ca-bundle ConfigMap with an invalid CA certificate.
+    """
+    ca_configmap_name = request.param.get("ca_configmap_name", "odh-trusted-ca-bundle")
+    ca_file_name = request.param.get("ca_file_name", "invalid-ca.crt")
+    ca_file_path = f"{CA_MOUNT_PATH}/{ca_file_name}"
+    ca_data = {ca_file_name: "-----BEGIN CERTIFICATE-----\nINVALIDCERTIFICATE\n-----END CERTIFICATE-----"}
+    ca_configmap = ConfigMap(
+        client=admin_client,
+        name=ca_configmap_name,
+        namespace=model_registry_namespace,
+        ensure_exists=True,
+    )
+    patch = {
+        "metadata": {
+            "name": ca_configmap_name,
+            "namespace": model_registry_namespace,
+        },
+        "data": ca_data,
+    }
+    with ResourceEditor(patches={ca_configmap: patch}):
+        yield ca_file_path
+
+
+@pytest.fixture(scope="class")
+def mysql_template_with_ca(model_registry_db_secret: Secret) -> dict[str, Any]:
+    """
+    Patches the MySQL template with the CA file path and volume mount.
+    Args:
+        model_registry_db_secret: The secret for the model registry's MySQL database
+    Returns:
+        dict[str, Any]: The patched MySQL template
+    """
+    mysql_template = get_model_registry_deployment_template_dict(
+        secret_name=model_registry_db_secret.name,
+        resource_name=DB_RESOURCES_NAME,
+    )
+    mysql_template["spec"]["containers"][0]["args"].append(f"--ssl-ca={CA_FILE_PATH}")
+    mysql_template["spec"]["containers"][0]["volumeMounts"].append({
+        "mountPath": CA_MOUNT_PATH,
+        "name": CA_CONFIGMAP_NAME,
+        "readOnly": True,
+    })
+    mysql_template["spec"]["volumes"].append({"name": CA_CONFIGMAP_NAME, "configMap": {"name": CA_CONFIGMAP_NAME}})
+    return mysql_template
+
+
+@pytest.fixture(scope="class")
+def deploy_secure_mysql_and_mr(
+    model_registry_namespace: str,
+    model_registry_db_secret: Secret,
+    model_registry_db_deployment: Deployment,
+    model_registry_mysql_config: dict[str, Any],
+    mysql_template_with_ca: dict[str, Any],
+    patch_mysql_deployment_with_ssl_ca: Deployment,
+) -> Generator[ModelRegistry, None, None]:
+    """
+    Deploy a secure MySQL and Model Registry instance.
+    Args:
+        model_registry_namespace: The namespace of the model registry
+        model_registry_db_secret: The secret for the model registry's MySQL database
+        model_registry_db_deployment: The deployment for the model registry's MySQL database
+        model_registry_mysql_config: The MySQL config dictionary
+        mysql_template_with_ca: The MySQL template with the CA file path and volume mount
+    """
+    with ModelRegistry(
+        name=SECURE_MR_NAME,
+        namespace=model_registry_namespace,
+        label=MODEL_REGISTRY_STANDARD_LABELS,
+        grpc={},
+        rest={},
+        istio=ISTIO_CONFIG_DICT,
```

Comment:
We don't need this anymore and should use oauth.

Reply:
you are right, thanks, I was still gettin MR `from ocp_resources.model_registry import ModelRegistry`